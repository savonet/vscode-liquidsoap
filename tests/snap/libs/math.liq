# Compute the minimum of two values.
# @category Math
def min(a, b)
  if a <= b then a else b end
end

# Compute the maximum of two values.
# @category Math
def max(a, b)
  if a >= b then a else b end
end

# Convert linear scale into decibels.
# @category Math
def dB_of_lin(x)
  20. * log10(x)
end

# Convert decibels into linear scale.
# @category Math
def lin_of_dB(x)
  pow(10., x / 20.)
end
