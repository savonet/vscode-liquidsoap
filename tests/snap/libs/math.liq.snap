># Compute the minimum of two values.
#^ source.liquidsoap comment.line.number-sign.liquidsoap
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># @category Math
#^ source.liquidsoap comment.line.number-sign.liquidsoap
# ^ source.liquidsoap comment.line.number-sign.liquidsoap
#  ^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap comment.doc.category.liquidsoap
#           ^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>def min(a, b)
#^^^ source.liquidsoap keyword.other.function.definition.liquidsoap
#   ^ source.liquidsoap
#    ^^^ source.liquidsoap entity.name.function.binding.liquidsoap
#       ^ source.liquidsoap
#        ^ source.liquidsoap variable.liquidsoap
#         ^ source.liquidsoap keyword.other.liquidsoap punctuation.comma punctuation.separator.comma
#          ^ source.liquidsoap
#           ^ source.liquidsoap variable.liquidsoap
#            ^ source.liquidsoap
>  if a <= b then a else b end
#^^ source.liquidsoap
#  ^^ source.liquidsoap keyword.control.liquidsoap
#    ^ source.liquidsoap
#     ^ source.liquidsoap variable.liquidsoap
#      ^ source.liquidsoap
#       ^^ source.liquidsoap keyword.operator.compare.liquidsoap
#         ^ source.liquidsoap
#          ^ source.liquidsoap variable.liquidsoap
#           ^ source.liquidsoap
#            ^^^^ source.liquidsoap keyword.control.liquidsoap
#                ^ source.liquidsoap
#                 ^ source.liquidsoap variable.liquidsoap
#                  ^ source.liquidsoap
#                   ^^^^ source.liquidsoap keyword.control.liquidsoap
#                       ^ source.liquidsoap
#                        ^ source.liquidsoap variable.liquidsoap
#                         ^ source.liquidsoap
#                          ^^^ source.liquidsoap keyword.control.liquidsoap
>end
#^^^ source.liquidsoap keyword.other.function.end.liquidsoap
>
># Compute the maximum of two values.
#^ source.liquidsoap comment.line.number-sign.liquidsoap
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># @category Math
#^ source.liquidsoap comment.line.number-sign.liquidsoap
# ^ source.liquidsoap comment.line.number-sign.liquidsoap
#  ^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap comment.doc.category.liquidsoap
#           ^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>def max(a, b)
#^^^ source.liquidsoap keyword.other.function.definition.liquidsoap
#   ^ source.liquidsoap
#    ^^^ source.liquidsoap entity.name.function.binding.liquidsoap
#       ^ source.liquidsoap
#        ^ source.liquidsoap variable.liquidsoap
#         ^ source.liquidsoap keyword.other.liquidsoap punctuation.comma punctuation.separator.comma
#          ^ source.liquidsoap
#           ^ source.liquidsoap variable.liquidsoap
#            ^ source.liquidsoap
>  if a >= b then a else b end
#^^ source.liquidsoap
#  ^^ source.liquidsoap keyword.control.liquidsoap
#    ^ source.liquidsoap
#     ^ source.liquidsoap variable.liquidsoap
#      ^ source.liquidsoap
#       ^^ source.liquidsoap keyword.operator.compare.liquidsoap
#         ^ source.liquidsoap
#          ^ source.liquidsoap variable.liquidsoap
#           ^ source.liquidsoap
#            ^^^^ source.liquidsoap keyword.control.liquidsoap
#                ^ source.liquidsoap
#                 ^ source.liquidsoap variable.liquidsoap
#                  ^ source.liquidsoap
#                   ^^^^ source.liquidsoap keyword.control.liquidsoap
#                       ^ source.liquidsoap
#                        ^ source.liquidsoap variable.liquidsoap
#                         ^ source.liquidsoap
#                          ^^^ source.liquidsoap keyword.control.liquidsoap
>end
#^^^ source.liquidsoap keyword.other.function.end.liquidsoap
>
># Convert linear scale into decibels.
#^ source.liquidsoap comment.line.number-sign.liquidsoap
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># @category Math
#^ source.liquidsoap comment.line.number-sign.liquidsoap
# ^ source.liquidsoap comment.line.number-sign.liquidsoap
#  ^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap comment.doc.category.liquidsoap
#           ^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>def dB_of_lin(x)
#^^^ source.liquidsoap keyword.other.function.definition.liquidsoap
#   ^ source.liquidsoap
#    ^^^^^^^^^ source.liquidsoap entity.name.function.binding.liquidsoap
#             ^ source.liquidsoap
#              ^ source.liquidsoap variable.liquidsoap
#               ^ source.liquidsoap
>  20. * log10(x)
#^^ source.liquidsoap
#  ^^ source.liquidsoap constant.numeric.decimal.integer.liquidsoap
#    ^^ source.liquidsoap
#      ^ source.liquidsoap keyword.operator.arithmetic.liquidsoap
#       ^ source.liquidsoap
#        ^^^^^ source.liquidsoap meta.function-call.liquidsoap entity.name.function.liquidsoap
#             ^ source.liquidsoap meta.function-call.liquidsoap
#              ^ source.liquidsoap meta.function-call.liquidsoap variable.liquidsoap
#               ^ source.liquidsoap meta.function-call.liquidsoap
>end
#^^^ source.liquidsoap keyword.other.function.end.liquidsoap
>
># Convert decibels into linear scale.
#^ source.liquidsoap comment.line.number-sign.liquidsoap
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># @category Math
#^ source.liquidsoap comment.line.number-sign.liquidsoap
# ^ source.liquidsoap comment.line.number-sign.liquidsoap
#  ^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap comment.doc.category.liquidsoap
#           ^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>def lin_of_dB(x)
#^^^ source.liquidsoap keyword.other.function.definition.liquidsoap
#   ^ source.liquidsoap
#    ^^^^^^^^^ source.liquidsoap entity.name.function.binding.liquidsoap
#             ^ source.liquidsoap
#              ^ source.liquidsoap variable.liquidsoap
#               ^ source.liquidsoap
>  pow(10., x / 20.)
#^^ source.liquidsoap
#  ^^^ source.liquidsoap meta.function-call.liquidsoap entity.name.function.liquidsoap
#     ^ source.liquidsoap meta.function-call.liquidsoap
#      ^^ source.liquidsoap meta.function-call.liquidsoap constant.numeric.decimal.integer.liquidsoap
#        ^ source.liquidsoap meta.function-call.liquidsoap
#         ^ source.liquidsoap meta.function-call.liquidsoap keyword.other.liquidsoap punctuation.comma punctuation.separator.comma
#          ^ source.liquidsoap meta.function-call.liquidsoap
#           ^ source.liquidsoap meta.function-call.liquidsoap variable.liquidsoap
#            ^ source.liquidsoap meta.function-call.liquidsoap
#             ^ source.liquidsoap meta.function-call.liquidsoap keyword.operator.arithmetic.liquidsoap
#              ^ source.liquidsoap meta.function-call.liquidsoap
#               ^^ source.liquidsoap meta.function-call.liquidsoap constant.numeric.decimal.integer.liquidsoap
#                 ^ source.liquidsoap meta.function-call.liquidsoap
#                  ^ source.liquidsoap meta.function-call.liquidsoap
>end
#^^^ source.liquidsoap keyword.other.function.end.liquidsoap
>